modin/experimental/engines/omnisci_on_ray/io.py

```diff
@@ -288,3 +343,127 @@ def _prepare_pyarrow_usecols(cls, read_csv_kwargs):
                 raise NotImplementedError("unsupported `usecols` parameter")
 
         return usecols_md
+
+    @classmethod
+    def _read_csv_check_support(
+        cls,
+        read_csv_kwargs: ReadCsvKwargsType,
+    ) -> Tuple[bool, str]:
+        """
+        Check if passed parameters are supported by current ``modin.pandas.read_csv`` implementation.
+
+        Parameters
+        ----------
+        read_csv_kwargs : dict
+                Parameters of read_csv function.
+
+        Returns
+        -------
+        bool
+            Whether passed parameters are supported or not.
+        str
+            Error message that should be raised if user explicitly set `engine="arrow"`.
+        """
+        filepath_or_buffer = read_csv_kwargs.get("filepath_or_buffer", None)
+        header = read_csv_kwargs.get("header", "infer")
+        names = read_csv_kwargs.get("names", None)
+        engine = read_csv_kwargs.get("engine", None)
+        skiprows = read_csv_kwargs.get("skiprows", None)
+        delimiter = read_csv_kwargs.get("delimiter", None)
+
+        if read_csv_kwargs.get("compression", "infer") != "infer":
+            return (
+                False,
+                "read_csv with 'arrow' engine doesn't support compression parameter, compression must be inferred"
+                " automatically (supported compression types are gzip and bz2)",
+            )
+
+        if isinstance(filepath_or_buffer, str):
+            if not os.path.exists(filepath_or_buffer):
+                from pandas.io.common import is_url
+
+                if cls.file_exists(filepath_or_buffer) or is_url(filepath_or_buffer):
+                    return (
+                        False,
+                        "read_csv with 'arrow' engine supports only local files",
+                    )
+                else:
+                    raise FileNotFoundError("No such file or directory")
+        elif not cls.pathlib_or_pypath(filepath_or_buffer):
+            if hasattr(filepath_or_buffer, "read"):
+                return (
+                    False,
+                    "read_csv with 'arrow' engine doesn't support file-like objects",
+                )
+            else:
+                raise ValueError(
+                    f"Invalid file path or buffer object type: {type(filepath_or_buffer)}"
+                )
+
+        read_csv_signature = inspect.signature(cls.read_csv)
+        read_csv_default_kwargs = {
+            k: v.default
+            for k, v in read_csv_signature.parameters.items()
+            if v.default is not inspect.Parameter.empty
+        }
```

Comment:
I suggest making this a class-level attribute computed after `read_csv` declaration, as this won't ever change - no need to do the same (potentially intensive) computation each time a user wants to read a file; you can also use `unsupported_args` to filter supported ones (effectively leaving only the unsupported subset of kwargs which have default values), this would lead to something like

```python
class OmniSciOnRay(...):
    # some class attrs here
    def read_csv(...):
        # read_csv body
    # somewhere in the class body
    read_csv_defaults = {k: v.default for k, v in inspect.signature(read_csv).parameters.items() if v.default is not inspect.Parameter.empty and k in unsupported_args}
    def _read_csv_check_support(...):
        # body
        for arg, def_value in cls.read_csv_defaults.items():
            if read_csv_kwargs[arg] != def_value:
                return False, "bad arg"
```

Reply:
Sounds reasonable, suggestion was implemented (`read_csv_defaults` dict comprehension was replaced with loop to avoid 
comprehension scope problem with `unsupported_args`).